pkg/limitador/k8s_objects.go

```diff
@@ -0,0 +1,145 @@
+package limitador
+
+import (
+	limitadorv1alpha1 "github.com/3scale/limitador-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	DefaultVersion   = "latest"
+	DefaultReplicas  = 1
+	ServiceName      = "limitador"
+	ServiceNamespace = "default"
+	Image            = "quay.io/3scale/limitador"
+	StatusEndpoint   = "/status"
+)
+
+func LimitadorService() *v1.Service {
+	return &v1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceName,
+			Namespace: ServiceNamespace,
+			Labels:    labels(),
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Name:       "http",
+					Protocol:   v1.ProtocolTCP,
+					Port:       8080,
+					TargetPort: intstr.FromString("http"),
+				},
+				{
+					Name:       "grpc",
+					Protocol:   v1.ProtocolTCP,
+					Port:       8081,
+					TargetPort: intstr.FromString("grpc"),
+				},
+			},
+			Selector:  labels(),
+			ClusterIP: v1.ClusterIPNone,
```

Comment:
Has this been changed since the review? I didn't pay attention to this field before. Is this intentional? Keep in mind `v1.ClusterIPNone` is different behavior than the empty string (or equivalently, unset value)

Reply:
It has not changed since the review.
For now, I'm just using the same values used in the Kubernetes templates of the Limitador repo: https://github.com/3scale-labs/limitador/blob/master/limitador-server/kubernetes/limitador-service.yaml

Reply:
ok,

Then I assume that's he behavior you want. In any case here's the info about the behavior of that field in case you are interested in details:
```
	// clusterIP is the IP address of the service and is usually assigned
	// randomly by the master. If an address is specified manually and is not in
	// use by others, it will be allocated to the service; otherwise, creation
	// of the service will fail. This field can not be changed through updates.
	// Valid values are "None", empty string (""), or a valid IP address. "None"
	// can be specified for headless services when proxying is not required.
	// Only applies to types ClusterIP, NodePort, and LoadBalancer. Ignored if
	// type is ExternalName.
	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
```

Reply:
The value is intended because a headless service is required for envoy's grpc client to load balance the connections correctly.

Reply:
:+1: